marahgh/assignment1/Q1.cpp

```diff
@@ -8,49 +7,54 @@ using namespace std;
 //My assumption:the words alphabet is english letters.    
 
 
-//---------------------------------------------------------------------------------------------------------
-/*A function that determines if two strings are anagrams ,
-	able to handle both case sensitive and case insensitive anagrams,
-	according to value of caseSensetive flag variable */
-bool areAnagrams(string s1, string s2, bool caseSensetive){
-    const int LEN= 'z' -'a' +1;
-
-    if(s1.size()!= s2.size()){
-        return false;
+/*auxiliary function for updating the given histogram, according to the value of the flag add
+  increments the counters if its true and decrements them otherwise*/
+bool updateHistogram(vector<int>& histogram,string word,bool add, bool caseSensetive){

```

Comment:
A note on interface design: Having a bool parameter can be confusing (is easy to think one way or the other, like it was true for add or true for substract). Having two in a row is double confusing :)